app/src/core/components/Modal.tsx

```diff
@@ -0,0 +1,60 @@
+import Dialog, { DialogProps } from '@mui/material/Dialog';
+import DialogContent from '@mui/material/DialogContent';
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { useTheme } from '@mui/material/styles';
+import { DialogActions, DialogTitle } from '@mui/material';
+import { ReactNode } from 'react';
+
+interface ResponsiveModalProps extends Omit<DialogProps, 'title'> {
+    /**
+     * Callback function to be called when the modal is closed
+     *
+     * @memberof ResponsiveModalProps
+     */
+    onCloseCallback?: () => void;
+    /**
+     * Disables the close button on the modal
+     *
+     * @memberof ResponsiveModalProps
+     */
+    closeDisabled?: boolean;
+    /**
+     * Action children to be rendered in the modal
+     *
+     * @memberof ResponsiveModalProps
+     */
+    actions?: ReactNode;
+    /**
+     * Overrides the default fullscreen behavior of showing full screen based upon the theme breakpoints (md or less)
+     *
+     * @memberof ResponsiveModalProps
+     */
+    fullScreen?: boolean;
+    /**
+     * Title of the modal
+     *
+     * @memberof ResponsiveModalProps
+     */
+    title?: ReactNode;
+}
+
+export default function Modal({ title, children, onCloseCallback, closeDisabled, actions, fullScreen, ...rest }: ResponsiveModalProps) {
+    const theme = useTheme();
+    const isFullScreen = fullScreen ?? useMediaQuery(theme.breakpoints.down('md'));
```

Comment:
not sure if you saw my slack thread comments on this, but i dont think you actually need to define `fullScreen` here or in your props interface, since `Dialog` already has it. Also this might technically be invalid in strict react rendering since hooks are not supposed to be conditionally invoked.

You should be able to remove all those changes and just do
```tsx
<Dialog fullScreen={fullScreen}  {...rest} 
```